homeassistant/components/media_player/firetv.py

```diff
@@ -4,166 +4,126 @@
-        self.port = port
-        self.device_id = device_id
-
-    @property
-    def state(self):
-        """Get the device state. An exception means UNKNOWN state."""
-        try:
-            response = requests.get(
-                DEVICE_STATE_URL.format(
-                    self.proto, self.host, self.port, self.device_id
-                    ), timeout=10).json()
-            return response.get('state', STATE_UNKNOWN)
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Could not retrieve device state for %s", self.device_id)
-            return STATE_UNKNOWN
-
-    @property
-    def current_app(self):
-        """Return the current app."""
-        try:
-            response = requests.get(
-                DEVICE_APPS_URL.format(
-                    self.proto, self.host, self.port, self.device_id, 'current'
-                    ), timeout=10).json()
-            _current_app = response.get('current_app')
-            if _current_app:
-                return _current_app.get('package')
-
-            return None
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Could not retrieve current app for %s", self.device_id)
-            return None
-
-    @property
-    def running_apps(self):
-        """Return a list of running apps."""
-        try:
-            response = requests.get(
-                DEVICE_APPS_URL.format(
-                    self.proto, self.host, self.port, self.device_id, 'running'
-                    ), timeout=10).json()
-            return response.get('running_apps')
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Could not retrieve running apps for %s", self.device_id)
-            return None
-
-    def action(self, action_id):
-        """Perform an action on the device."""
-        try:
-            requests.get(DEVICE_ACTION_URL.format(
-                self.proto, self.host, self.port, self.device_id, action_id
-                ), timeout=10)
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Action request for %s was not accepted for device %s",
-                action_id, self.device_id)
-
-    def start_app(self, app_name):
-        """Start an app."""
-        try:
-            requests.get(DEVICE_APPS_URL.format(
-                self.proto, self.host, self.port, self.device_id,
-                app_name + '/start'), timeout=10)
-        except requests.exceptions.RequestException:
-            _LOGGER.error(
-                "Could not start %s on %s", app_name, self.device_id)
+    adbkey = config.get(CONF_ADBKEY)
+    get_source = config.get(CONF_GET_SOURCE)
+    get_sources = config.get(CONF_GET_SOURCES)
+
+    device = FireTVDevice(host, name, adbkey, get_source, get_sources)
+    adb_log = " using adbkey='{0}'".format(adbkey) if adbkey else ""
+    if not device._firetv._adb:
```

Comment:
If you want to access the internals of the `FireTVDevice`, you should probably add a public getter (or a property) for those, or provide a property shortcut for the wanted features (`is_available`?). That was one of those complaints from linter.

Reply:
You mean something like this?

```python
@property
def adb(self):
    return self._firetv._adb
```

Reply:
Yes, something like that would work (and make the code cleaner).

Reply:
There are 4 hidden attributes of `self._firetv` that I access: `_adb`, `_screen_on`, `_awake`, and `_wake_lock`. Should I define a property for getting each of them, plus a method for setting the `_adb` attribute? It just seems cleaner to me to leave it as it is.

Reply:
Fixed in https://github.com/JeffLIrion/home-assistant/commit/91b7b9b73ae68dd2c0cf087a53d3efc1a89258be (requires firetv 1.0.7 to be released on pypi)